pandas/io/sql.py

```diff
@@ -2769,10 +2796,11 @@ def to_sql(
         frame: DataFrame
         name: string
             Name of SQL table.
-        if_exists: {'fail', 'replace', 'append'}, default 'fail'
+        if_exists: {'fail', 'replace', 'append', 'delete_rows'}, default 'fail'
             fail: If table exists, do nothing.
             replace: If table exists, drop it, recreate it, and insert data.
             append: If table exists, insert data. Create if it does not exist.
+            delete_rows: If a table exists, delete all records and insert data.
```

Comment:
Can you add a warning to this docstring that pandas does not sanitize the schema and table name as part of a delete operation? You can use our warning about DataFrame.query as a reference:

https://pandas.pydata.org/docs/dev/reference/api/pandas.DataFrame.query.html

The underlying ADBC drivers are doing this now, but I don't know that there is a guarantee every implementation will in the future. And I'm not sure about the SQLAlchemy landscape...so want to make it clear to end users that pandas is not adding any security features on top of that. They need to be careful